int32 adrs
int32 pote
bool  flag
